pkg/providers/az-client.go

```diff
@@ -13,3 +17,26 @@ type AzClient struct {
 type azTenantClient interface {
 	NewListPager(options *armsubscription.TenantsClientListOptions) *runtime.Pager[armsubscription.TenantsClientListResponse]
 }
+
+type GraphClient interface {
+	GetApplicationObjectId(ctx context.Context, appId string) (string, error)
+}
+
+// GraphServiceClientImpl implements the GraphClient interface.
+type GraphServiceClientImpl struct {
+	Client *msgraph.GraphServiceClient
+}
+
+func (g *GraphServiceClientImpl) GetApplicationObjectId(ctx context.Context, appId string) (string, error) {
```

Comment:
is there a [type assertion](https://go.dev/tour/methods/15) for this interface? otherwise we don't explicitly constrain the struct to it

Reply:
this isn't the kind of type assertion you mean right?

you mean the `var _ GraphService = &GraphServiceClientImpl{}` kind?

Reply:
yes you're correct i mean the [type satisfies the interface assertion](https://stackoverflow.com/questions/10498547/ensure-a-type-implements-an-interface-at-compile-time-in-go#:~:text=In%20the%20Go%20language%20there%20is%20no%20%22,one%20in%20the%20assignment%20check%21%20type%20T%20struct%7B%7D)

Reply:
I have used var _ GraphClient = (*GraphServiceClient)(nil) - type satisfies an interface without creating an actual instance of the type. Let me know if this is ok.